var/spack/repos/builtin/packages/verrou/package.py

```diff
@@ -0,0 +1,110 @@
+##############################################################################
+from spack import *
+import os
+import subprocess
+import sys
+
+
+class Verrou(AutotoolsPackage):
+    """A floating-point error checker.
+
+    Verrou helps you look for floating-point round-off errors in programs. It
+    implements a stochastic floating-point arithmetic based on random rounding:
+    all floating-point operations are perturbed by randomly switching rounding
+    modes. This can be seen as an asynchronous variant of the CESTAC method, or
+    a subset of Monte Carlo Arithmetic, performing only output randomization
+    through random rounding.
+    """
+
+    homepage = "https://github.com/edf-hpc/verrou"
+    url      = "https://github.com/edf-hpc/verrou.git"
+
+    version('master',
+            git='https://github.com/edf-hpc/verrou.git',
+            branch='master')
+
+    version('2.0.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='a614ad695d060abdfcd3cec11022d6f044a0980d')
+    version('1.1.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='d69787e89507519e14f5ec44150c1882930f692e')
+    version('1.0.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='519dd9a83738f72da8446322e9c403e608940b7a')
+    version('0.9.0',
+            git='https://github.com/edf-hpc/verrou.git',
+            commit='ebabfcf7f4e9a02b130e9819ba6c19e0450eb73d')
+
+    resource(name='valgrind-3.13.0',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='1378ec95f22235e0a8c972cf1cd0abef0b9610d4',
+             when='@1.1:,master')
+    resource(name='valgrind-3.12.0',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='36923ab298c8d2791d6c86b6cd5c0667c07449f6',
+             when='@1.0')
+    resource(name='valgrind-3.10.1',
+             git='git://sourceware.org/git/valgrind.git',
+             commit='cfc3175a6eb132a5600fcf705f3fa326d71483bf',
+             when='@0.9')
+
+    conflicts('valgrind')
+
+    variant('fma', default=True,
+            description='Activates fused multiply-add support for Verrou')
+
+    def patch(self):
+        # The current setup gives us the verrou source tree, with a "valgrind"
+        # subdirectory. But we want the reverse layout. Let's fix this.
+        verrou_files = os.listdir('.')
+        verrou_files.remove('valgrind')
+        os.mkdir('verrou')
+        for name in verrou_files:
+            os.rename(name, os.path.join('verrou', name))
+        for name in os.listdir('valgrind'):
+            os.rename(os.path.join('valgrind', name), name)
+        os.rmdir('valgrind')
+
+        # Once this is done, we can patch valgrind
+        with open('verrou/valgrind.diff') as patchfile:
+            subprocess.call(['patch', '-p0'], stdin=patchfile)
+
+    def autoreconf(self, spec, prefix):
+        # Needed because we patched valgrind
+        which("bash")("autogen.sh")
```

Comment:
You'll probably need to add the following build dependencies:

- m4
- libtool 
- automake 
- autoconf

Reply:
I initially added these, then removed them when I realized that the valgrind package, which this work is partially based on, only enabled them for the develop branch. Will experiment with adding them back...

Reply:
...and here I am bitten by the Linux shebang line length limit...

Reply:
Well, from further investigation, it looks like unlike during my previous attempt at building this package, I do not need to manually run autogen. This is strange... but it also means that I can probably live without explicit dependencies on automake/autoconf for now.